File: 2021/test/day_10_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_10"

class Day10Test < Minitest::Test
  def test_part1
    day = Day10.new

    assert_equal 26_397, day.part1(input("day_10_example_1"))
    assert_equal 265_527, day.part1(input("day_10"))
  end

  def test_part2
    day = Day10.new

    assert_equal 288_957, day.part2(input("day_10_example_1"))
    assert_equal 3_969_823_589, day.part2(input("day_10"))
  end
end

File: 2025/test/day_10_test.exs
defmodule Day10Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day10.part_1(input("day_10_example_1")) == 7
    assert Day10.part_1(input("day_10")) == 486
  end

  # test "part2" do
  #   assert Day09.part_2(input("day_09_example_1")) == 40
  #   assert Day09.part_2(input("day_09")) == 15_650_261_281_478
  # end
end
